sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/TransientIOErrorsRetryingIterator.scala

```diff
@@ -112,7 +118,31 @@ private class TransientIOErrorsRetryingIterator[TSparkRow]
           currentFeedResponseIterator.get
       }
 
-      if (feedResponseIterator.hasNext) {
+      val hasNext: Boolean = try {
+        Await.result(
+          Future {
+            feedResponseIterator.hasNext
```

Comment:
why not use e2e timeout config on the queryItems call?

Reply:
Fixed as discussed offline